site/src/pages/WorkspacePage/WorkspaceDeleteDialog/WorkspaceDeleteDialog.tsx

```diff
@@ -0,0 +1,182 @@
+  onCancel: () => void;
+  onConfirm: (arg: CreateWorkspaceBuildRequest["orphan"]) => void;
+}
+export const WorkspaceDeleteDialog = (props: WorkspaceDeleteDialogProps) => {
+  const { workspace, canUpdateTemplate, isOpen, onCancel, onConfirm } = props;
+
+  const hookId = useId();
+  const [userConfirmationText, setUserConfirmationText] = useState("");
+  const [orphanWorkspace, setOrphanWorkspace] =
+    useState<CreateWorkspaceBuildRequest["orphan"]>(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const deletionConfirmed = workspace.name === userConfirmationText;
+  const onSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    if (deletionConfirmed) {
+      onConfirm(orphanWorkspace);
+    }
+  };
+
+  const hasError = !deletionConfirmed && userConfirmationText.length > 0;
+  const displayErrorMessage = hasError && !isFocused;
+  const inputColor = hasError ? "error" : "primary";
+
+  return (
+    <ConfirmDialog
+      type="delete"
+      hideCancel={false}
+      open={isOpen}
+      title="Delete Workspace"
+      onConfirm={() => onConfirm(orphanWorkspace)}
+      onClose={onCancel}
+      disabled={!deletionConfirmed}
+      description={
+        <>
+          <div css={styles.workspaceInfo}>
+            <div css={{ flexDirection: "column" }}>
+              <p className="name">{workspace.name}</p>
+              <p className="label">workspace</p>
+            </div>
+            <div css={{ flexDirection: "column" }}>
+              <p className="info">{dayjs(workspace.created_at).fromNow()}</p>
+              <p className="label">created</p>
+            </div>
+          </div>
+
+          <p>Deleting this workspace is irreversible!</p>
+          <p>
+            Type &ldquo;<strong>{workspace.name}</strong>&ldquo; below to
+            confirm:
+          </p>
+
+          <form onSubmit={onSubmit}>
+            <TextField
+              fullWidth
+              autoFocus
+              css={{ marginTop: 32 }}
+              name="confirmation"
+              autoComplete="off"
+              id={`${hookId}-confirm`}
+              placeholder={workspace.name}
+              value={userConfirmationText}
+              onChange={(event) => setUserConfirmationText(event.target.value)}
+              onFocus={() => setIsFocused(true)}
+              onBlur={() => setIsFocused(false)}
+              label="Workspace name"
+              color={inputColor}
+              error={displayErrorMessage}
+              helperText={
+                displayErrorMessage &&
+                `${userConfirmationText} does not match the name of this workspace`
+              }
+              InputProps={{ color: inputColor }}
+              inputProps={{
+                "data-testid": "delete-dialog-name-confirmation",
+              }}
+            />
+            {canUpdateTemplate && (
+              <div css={styles.orphanContainer}>
+                <div css={{ flexDirection: "column" }}>
```

Comment:
Does this rule do anything?

Since this is a child of the `orphanContainer` div, I'm expecting that the `flexDirection` rule has no effect, because the element would go back to being a block

Reply:
Same concern for the div that comes a few lines later

Reply:
:D oh flexbox